include/arch/riscv/arch/machine.h

```diff
@@ -117,6 +117,9 @@ static inline void setVSpaceRoot(paddr_t addr, asid_t asid)
                            asid,                         /* asid */
                            addr >> seL4_PageBits); /* PPN */
 
+    /* Make sure are reads/writes are ordered before writing sptbr*/
+    sfence();
```

Comment:
Even though this is just one line, but I'm sure @chrisguikema is aware how vital it's to run on actual HW. @pingerino  I'm not sure when will I get the CLA, so I'm happy with @chrisguikema authoring it not to block this PR

Reply:
@heshamelmatary Thank you for the feedback. I'll make the same commit and mention you in the commit message.

For future reference, you can run a single thread on actual hardware without this change. Once you add a second thread, I noticed that the second thread would run for one tick, and then the system would fault into the `idle` thread.

Reply:
Correct, I have debugged this issue for a while. To be more specific, it happens when having more than one thread with different ASIDs, after the first context switch. That's a TLB inconsistency issue. The riscv-priv manual suggests it *might* be needed before writing to ```satp``` register. I've had the impression that having an ```sfence``` after each ```[un]map``` and ```remap```  operation that should be sufficient, but that's not the case on Rocket